stdlib/LinearAlgebra/src/bunchkaufman.jl

```diff
@@ -4,6 +4,15 @@
 ## LD for BunchKaufman, UL for CholeskyDense, LU for LUDense and
 ## define size methods for Factorization types using it.
 
+##----------- Type utilities for generic Bunch-Kaufman implementation ------------
+# Generic real type. Any real number type should able to approximate
+# real numbers, and thus be closed under arithmetic operations.
+# Therefore so Int, Complex{Int}, etc. are excluded.
+ClosedReal = T where T <: Union{AbstractFloat, Rational}
```

Comment:
Why not just use `Real` here?

Reply:
Basically to exclude `Irrational`. As far as I know, irrationals in Julia are not closed under arithmetic operations, so `π+1` does not return a number of type `Irrational`. I don't how to process irrationals in a type stable fashion, so I was forced to exclude them.